### PR TITLE
test: add test file for deleteAgendaFolder.ts

### LINKED ISSUE #3825: Test: src/graphql/types/Mutation/deleteAgendaFolder.ts
# Goal

The goal of this issue is to improve the code coverage for this file.

# Tasks

1. Review the file to identify sections of code that are being ignored by Codecov or are not covered by tests.
2. Create or update test cases to ensure 100% coverage for the file.
3. Remove any `/* istanbul ignore */` or equivalent statements that bypass code coverage reporting, unless absolutely necessary.

# Resources

1. Refer to the foundational documentation on writing test cases in the repository.
1. Check the [Codecov report](https://app.codecov.io/gh/PalisadoesFoundation/talawa-api/tree/develop/src?displayType=list) for details on the uncovered lines.
1. Read our documentation on mock isolation and running concurrent tests.
    - https://docs-api.talawa.io/docs/developer-resources/testing-validation/

1. Read our contributor guide. 
    - https://developer.palisadoes.org/docs/contributor-guide/contributing

1. Read our AI use policy. 
    - https://developer.palisadoes.org/docs/contributor-guide/ai
    
# Acceptance Criteria

1. All sections of the file are covered by tests.
1. Code coverage for the file reaches 100%.
1. PR created with necessary updates, passing all checks and reviews.
1. All tests must be valid and likely and edge cases covered.

### PR DESCRIPTION
**Pull request by abhishekblue**
**What kind of change does this PR introduce?**

Test

**Issue Number:**

Fixes #3825 

**Summary**

Added test file for `deleteAgendaFolder.ts` to improve test coverage.

**Does this PR introduce a breaking change?**

No

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [x] I have written tests for all new changes/features
- [x] I have verified that test coverage meets or exceeds 95%
- [x] I have run the test suite locally and all tests pass

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

No

## Summary by CodeRabbit

* **Tests**
  * Added extensive unit tests for agenda-folder deletion covering authentication, input validation, resource existence, authorization paths, successful deletes (with/without parent folder), and edge-case error handling to improve reliability and prevent regressions.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: test/graphql/types/Mutation/deleteAgendaFolder.test.ts [hunk 1]
```diff
@@ -0,0 +1,656 @@
...
+				ctx,
+			);
+
+			expect(memberEqMock).toHaveBeenCalledWith(
+				"memberships.memberId",
+				"user123",
+			);
+		});
+	});
+});
```
[the remark this answers is not in the document]
Comment (abhishekblue) [reply test/graphql/types/Mutation/deleteAgendaFolder.test.ts]: Why do we need to use mercuriusClient? Doesn’t this repository already use vi.mock for writing tests?

File: test/graphql/types/Mutation/deleteAgendaFolder.test.ts [hunk 2]
```diff
@@ -0,0 +1,884 @@
...
+import { createRegularUserUsingAdmin } from "../createRegularUserUsingAdmin";
+import {
+	Mutation_createAgendaFolder,
+	Mutation_createEvent,
+	Mutation_createOrganization,
+	Mutation_createOrganizationMembership,
+	Mutation_deleteOrganization,
+	Mutation_deleteStandaloneEvent,
+	Query_signIn,
+} from "../documentNodes";
```
[the remark this answers is not in the document]
Comment (abhishekblue) [reply test/graphql/types/Mutation/deleteAgendaFolder.test.ts:28]: Biome check failed even after updating the imports' sorting. Can you check again?

**Comment by palisadoes** [comment]
@coderabbitai full review

**Comment by abhishekblue** [comment]
@palisadoes  I see the branch was merged but its not merged yet. Could you please let me know the next steps for getting this merged?
